File: sdk/python/src/agent/position_int.py
class PositionInt:
    def __init__(self, x: int, y: int, angle: float):
        self.x = x
        self.y = y
        self.angle = angle

    def __str__(self):
        return f"Position(x={self.x}, y={self.y}, angle={self.angle})"

File: server/src/Protocol/Scheme/PositionInt.cs
using System.Text.Json.Serialization;

namespace Thuai.Server.Protocol.Scheme;

public record PositionInt
{
    [JsonPropertyName("x")]
    public required int X { get; init; }

    [JsonPropertyName("y")]
    public required int Y { get; init; }

    [JsonPropertyName("angle")]
    public required int Angle { get; init; }
}
